cmd/kubeadm/app/util/config/masterconfig.go

```diff
@@ -164,25 +165,38 @@ func BytesToInternalConfig(b []byte) (*kubeadmapi.InitConfiguration, error) {
 		}
 
 		if kubeadmutil.GroupVersionKindsHasInitConfiguration(gvk) {
-			if err := runtime.DecodeInto(kubeadmscheme.Codecs.UniversalDecoder(), fileContent, internalcfg); err != nil {
+			if err := runtime.DecodeInto(kubeadmscheme.Codecs.UniversalDecoder(), fileContent, initcfg); err != nil {
+				return nil, err
+			}
+			initConfigFound = true
+			continue
+		}
+		if kubeadmutil.GroupVersionKindsHasClusterConfiguration(gvk) {
+			if err := runtime.DecodeInto(kubeadmscheme.Codecs.UniversalDecoder(), fileContent, clustercfg); err != nil {
 				return nil, err
 			}
-			masterConfigFound = true
+			clusterConfigFound = true
 			continue
 		}
 
 		fmt.Printf("[config] WARNING: Ignored YAML document with GroupVersionKind %v\n", gvk)
 	}
+	// If InitConfiguration was found but not ClusterConfiguration, do nothing
+	if initConfigFound && !clusterConfigFound {	}
+	// If ClusterConfiguration was found but not ClusterConfiguration, just set 
+	if (!initConfigFound && clusterConfigFound) || (initConfigFound && clusterConfigFound) {
+		initcfg.ClusterConfiguration = *clustercfg
+	}
 	// Just as an extra safety check, don't proceed if a InitConfiguration object wasn't found
-	if !masterConfigFound {
-		return nil, fmt.Errorf("no InitConfiguration kind was found in the YAML file")
+	if !initConfigFound && !clusterConfigFound {
+		return nil, fmt.Errorf("no InitConfiguration or ClusterConfiguration kind was found in the YAML file")
```

Comment:
English here could be more exact with "Neither InitConfiguration nor ClusterConfiguration ..."